runtime/src/consensus/tendermint/verifier.rs

```diff
@@ -795,6 +895,13 @@ impl Verifier {
             "latest_height" => cache.latest_known_height(),
         );
 
+        // Verify state freshness using the host. This step is required only for clients
+        // as executors and key managers verify freshness regularly using node registration
+        // (RAK with random nonces).
+        if self.protocol.get_rak().is_none() {
```

Comment:
This won't work as the runtime will generate a RAK for Remote Attestation purposes in any case. It would be better if there was a configuration option that would enable this feature, e.g. in:
https://github.com/oasisprotocol/oasis-core/blob/b64e8b5a08c27a31066b47e232b09087b8c51f70/runtime/src/config.rs#L4-L18